components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -222,13 +215,28 @@ private fun load(
             loaders.forEach { loader ->
                 val result = loader.load(context, request, resource)
 
-                if (result != IconLoader.Result.NoResult) {
-                    return Pair(result, resource)
+                val icon = decodeIcon(result, decoders, desiredSize)
+
+                if (icon != null) {
+                    return icon to resource
                 }
             }
         }
 
-    return Pair(IconLoader.Result.NoResult, null)
+    return null
+}
+
+private fun decodeIcon(
```

Comment:
minor naming nit: We now have a `decodeIcon()` and a `decode()` function here. I feel like it would be more consistent if both were either the same (e.g. `decode()`) or if they are more explicit (`decodeIconLoaderResult()` vs. `decodeBytes()`).